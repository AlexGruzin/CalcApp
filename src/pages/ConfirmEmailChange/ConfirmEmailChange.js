import React from 'react';
import PropTypes from 'prop-types';
import { CONFIRMATION_STATUS } from 'constants/auth';
import Spinner from 'components/Spinner';
import { useTranslation } from 'react-i18next';

import './ConfirmEmailChange.styles.less';

const ConfirmEmailChange = ({ emailChangeStatus }) => {
  const { t } = useTranslation();

  return (
    <div className="email-chn">
      {!emailChangeStatus && (
        <>
          <Spinner />
          <div className="email-chn__text">{t('confirm:email.processing')}</div>
        </>
      )}
      {emailChangeStatus === CONFIRMATION_STATUS.SUCCESS && (
        <div className="email-chn__done">{t('confirm:success')}</div>
      )}
      {emailChangeStatus === CONFIRMATION_STATUS.ERROR && <div className="email-chn__block">{t('confirm:error')}</div>}
    </div>
  );
};

ConfirmEmailChange.propTypes = {
  emailChangeStatus: PropTypes.string,
};

ConfirmEmailChange.defaultProps = {
  emailChangeStatus: '',
};

export default ConfirmEmailChange;
