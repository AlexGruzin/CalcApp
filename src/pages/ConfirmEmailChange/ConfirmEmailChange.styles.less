.email-chn {
  display: flex;
  flex-grow: 1;
  background: var(--white);
  justify-content: center;
  align-items: center;
}

.email-chn__status {
  position: absolute;
}

.email-chn__text {
  margin-top: var(--space-x10);
  color: var(--font-black);
}

.email-chn .spinner > div {
  border-color: var(--black);
  border-right-color: transparent;
}

.email-chn__block {
  color: var(--error-color);
}

.email-chn__done {
  color: var(--turquoise);
}
