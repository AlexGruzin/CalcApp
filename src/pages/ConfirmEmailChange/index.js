import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { getEmailChangeStatus } from 'domains/auth/selectors';
import ConfirmEmailChange from './ConfirmEmailChange';

const selectors = createStructuredSelector({
  emailChangeStatus: getEmailChangeStatus,
});

export default connect(selectors, null)(ConfirmEmailChange);
