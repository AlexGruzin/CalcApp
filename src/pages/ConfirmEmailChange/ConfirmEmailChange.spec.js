import React from 'react';
import { render } from '@testing-library/react';
import ConfirmEmailChange from './ConfirmEmailChange';

describe('ConfirmEmailChange tests', () => {
  it('Check snapshots of ConfirmEmailChange are equal.', () => {
    const { asFragment } = render(<ConfirmEmailChange />);
    expect(asFragment()).toMatchSnapshot();
  });
});
