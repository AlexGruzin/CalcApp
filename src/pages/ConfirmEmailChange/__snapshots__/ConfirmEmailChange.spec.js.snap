// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailChange tests Check snapshots of ConfirmEmailChange are equal. 1`] = `
<DocumentFragment>
  <div
    class="email-chn"
  >
    <div
      class="spinner"
    >
      <div
        class="circle__Wrapper-sc-16bbsoy-0 dpSNSD"
        color="#fff"
        size="24"
      />
    </div>
    <div
      class="email-chn__text"
    >
      confirm:email.processing
    </div>
  </div>
</DocumentFragment>
`;
